tools/runners/run-mongo.js

```diff
@@ -213,8 +213,7 @@ if (process.platform === 'win32') {
             var foundPort = parseInt(m[2], 10);
             var foundPath = m[3];
 
-            if ( (! port || port === foundPort) &&
-                 (! dbDir || dbDir === foundPath)) {
+            if (! dbDir || dbDir === foundPath) {
```

Comment:
Also, shouldn't we make the same change on line 134 (the windows version). Although looking at that code, I don't see how it works in case 1. above either (it has the reverse problem).